extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -307,4 +315,64 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
         return new HealthBuildItem("io.quarkus.agroal.runtime.health.DataSourceHealthCheck",
                 agroalBuildTimeConfig.healthEnabled, "datasource");
     }
+
+    // TODO: introduce more metrics
+    @BuildStep
+    void registerMetrics(AgroalRuntimeConfig agroalRuntimeConfig,
```

Comment:
I need to inject the runtime config, because the build config doesn't contain the `metrics-enabled` property which enables/disables metric collection, and we need to work with its value here.